sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -565,6 +565,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
       filters: Array[Filter],
       inputPaths: Array[String],
       broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
Yeah, I agree. Basically it's impossible to

1. Create a temporary JSON table pointing to path `P`
2. Change the contents by arbitrary means without notifying the temporary table
3. Read the table again and expect to get updated contents

In the old JSON relation implementation, the refreshing logic is done by `TextInputFormat.listStatus()`, while the new `JSONRelation` relies on `HadoopFsRelation`. We can use `SqlNewHadoopRDD` and override the input format there to inject the `FileStatus` cache to avoid extra refreshing costs there.